### ISSUE TITLE
Easee energy use off by 1 hour

### ISSUE THREAD

**Issue by degerrit** [nordicopen/easee_hass#535]
### Question

I'm using the Easee integration with Home Assistant and have had few problems; but looking at the Energy dashboard I noticed something weird: the Easee energy use is off by one hour.

I'm in timezone CET (so UTC+1). I also had a look back at older energy graphs during CEST summer time (UTC+2) and it's the same: off by one hour. Home Assistant is displaying the non-Easee data timepoints correctly, but of course most of those are ingested in real-time, not at periodic intervals like I assume Easee data is.

So now I'm wondering: is there a configuration parameter that determines the timezone of the ingested Easee data - either in the integration or at the Easee end?

### What version of the integration are you using?

0.9.66

### Additional information

Energy graph 2024-12-05 (CET, Winter time, e.g. UTC+1): _you can see (based on the overall energy consumption data provided by the digital meter) that the Easee actually started charging between 22:00 and 23:00. But the energy entity for Easee only logs the start of the charge in the hour between 23:00 and 0:00:_
![HA easee off by 1hr 2024-12-05](https://github.com/user-attachments/assets/f2c2e3aa-1151-47db-96f8-f2f4b69ee9ee)

Energy graph 2024-10-05 (CEST, Summer time, e.g. UTC+2): _same pattern, here both start and stop time are on the graph, showing that the opposite occurs at stop time: because Easee reports energy usage for the hour that charging has in fact already stopped, there is a huge negative value. Interesting is that the chart is not off by +2 here, but still off by +1, like in the winter._
![HA easee off by 1hr CEST 2024-10-05](https://github.com/user-attachments/assets/4c123c1a-f9df-4a54-add3-11a503fd01fb)

**Comment by olalid** [comment]
See https://github.com/nordicopen/easee_hass/issues/462 and https://github.com/nordicopen/easee_hass/issues/387

**Comment by degerrit** [comment]
So i created a "Powercalc" Energy sensor ("Energy from real power sensor") based on the `sensor.easee_NAME_power` sensor, this "workaround" works just fine for me now - problem solved!

* https://docs.powercalc.nl/sensor-types/real-power-sensor/

I guess this issue can be closed. 

But I bet you will keep seeing these "bug reports" as more and more people try to use both the HA Energy feature and their charger data. 

Maybe it is better to make the Easee _energy use_ sensor _disabled-by-default_, and somehow annotate or rename it, so that people's expectations are correct. The name "energy_per_hour" kind of creates that expectation that also the hour component will be correct.

**Comment by olalid** [comment]
Sounds like you have misunderstood something here....
The energy sensor for the charger is called "lifetime_energy".
The energy per hour sensor that you are referring shows the total energy in the previous hour.
Since it collects statistics it always delayed by one hour since data for the full hour needs to be collected before it is possible to present it.
As such, the energy per hour sensor is useless as input to the energy panel, that is just not its purpose.

**Comment by olalid** [comment]
See https://github.com/nordicopen/easee_hass/pull/545

**Comment by degerrit** [comment]
>... The energy per hour sensor that you are referring shows the total energy in the previous hour. Since it collects statistics it always delayed by one hour since data for the full hour needs to be collected before it is possible to present it. As such, the energy per hour sensor is useless as input to the energy panel, that is just not its purpose.

Yes, exactly, that's how I understood it based on your first issue links :-) I'm closing this one (since my Energy dashboard Easee issue solved by using PowerCalc instead), I think/hope #545 will reduce the amount of bug reports unnecessarily going your way about this.
